main.go

```diff
@@ -19,253 +19,98 @@ package main
-	showLatestFlag := getopt.BoolLong("show-latest", 'U', "Show latest stable version")
-	mirrorURL := getopt.StringLong("mirror", 'm', defaultMirror, "Install from a remote API other than the default. Default: "+defaultMirror)
-	chDirPath := getopt.StringLong("chdir", 'c', dir, "Switch to a different working directory before executing the given command. Ex: tfswitch --chdir terraform_project will run tfswitch in the terraform_project directory")
-	versionFlag := getopt.BoolLong("version", 'v', "Displays the version of tfswitch")
-	defaultVersion := getopt.StringLong("default", 'd', defaultLatest, "Default to this version in case no other versions could be detected. Ex: tfswitch --default 1.2.4")
-	helpFlag := getopt.BoolLong("help", 'h', "Displays help message")
-	_ = versionFlag
 
-	getopt.Parse()
-	args := getopt.Args()
-
-	homedir, err := homedir.Dir()
-	if err != nil {
-		logger.Fatalf("Unable to get home directory: %v", err)
-		os.Exit(1)
-	}
-
-	TFVersionFile := filepath.Join(*chDirPath, tfvFilename)    //settings for .terraform-version file in current directory (tfenv compatible)
-	RCFile := filepath.Join(*chDirPath, rcFilename)            //settings for .tfswitchrc file in current directory (backward compatible purpose)
-	TOMLConfigFile := filepath.Join(*chDirPath, tomlFilename)  //settings for .tfswitch.toml file in current directory (option to specify bin directory)
-	HomeTOMLConfigFile := filepath.Join(homedir, tomlFilename) //settings for .tfswitch.toml file in home directory (option to specify bin directory)
-	TGHACLFile := filepath.Join(*chDirPath, tgHclFilename)     //settings for terragrunt.hcl file in current directory (option to specify bin directory)
+	parameters := param_parsing.GetParameters()
+	//defaults := lib.GetDefaults()
 
 	switch {
-	case *versionFlag:
-		//if *versionFlag {
-		logger.Infof("Version: %s", version)
-	case *helpFlag:
-		//} else if *helpFlag {
-		usageMessage()
-	/* Checks if the .tfswitch.toml file exist in home or current directory
-	 * This block checks to see if the tfswitch toml file is provided in the current path.
-	 * If the .tfswitch.toml file exist, it has a higher precedence than the .tfswitchrc file
-	 * You can specify the custom binary path and the version you desire
-	 * If you provide a custom binary path with the -b option, this will override the bin value in the toml file
-	 * If you provide a version on the command line, this will override the version value in the toml file
-	 */
-	case fileExists(TOMLConfigFile) || fileExists(HomeTOMLConfigFile):
-		version := ""
-		binPath := *custBinPath
-		if fileExists(TOMLConfigFile) { //read from toml from current directory
-			version, binPath = getParamsTOML(binPath, *chDirPath)
-		} else { // else read from toml from home directory
-			version, binPath = getParamsTOML(binPath, homedir)
-		}
-
-		switch {
-		/* GIVEN A TOML FILE, */
+	case parameters.VersionFlag:
+		fmt.Printf("\nVersion: %v\n", version)
```

Comment:
I'd really be keen to drop the leading empty line. WDYT?
```suggestion
		fmt.Printf("Version: %v\n", version)
```
Also why wouldn't print string as string instead of allowing arbitrary type via `%v`? I think we should ensure version is string and fail otherwise.
```suggestion
		fmt.Printf("Version: %s\n", version)
```